src/Bundle/Resources/config/services.xml

```diff
@@ -40,7 +40,7 @@
 
         <service id="sylius.email_renderer.adapter.abstract" class="Sylius\Component\Mailer\Renderer\Adapter\AbstractAdapter" abstract="true">
             <call method="setEventDispatcher">
-                <argument type="service" id="event_dispatcher" />
+                <argument type="service" id="event_dispatcher" on-invalid="ignore" />
```

Comment:
This (and the change immediately below) ends up being a bug fix.  The event dispatcher is an optional dependency in the classes that support it, yet compiling the container would fail hard if the dispatcher service isn't there.  This ensures that the dispatcher stays optional.

Reply:
It’s the “ignoring missing dependencies” section on that page.

The null behavior is valid if you’re working with a nullable argument.  The `$dispatcher` argument on the abstract adapter’s `setEventDispatcher()` method is not nullable, so the invalid behavior is the right behavior for this case (which means it just won’t be called).

Reply:
oh yes, I just had not seen it was  with a call method, excuxe me.